### PR TITLE
adding frequent positions to ai legs TM-3233

### PR DESCRIPTION
**Pull request by SidelineCory24**
[Mock PR](https://github.com/MetaPhase-Consulting/mock-fsbid/pull/321)

To-do

- [x] clicking "+" on a Frequent Position adds that position to the ai leg
- [x] a frequent position can't be added once 10 legs exist on the ai
- [x] can search for a Frequent Position

### CODE REVIEW

File: src/Components/Agenda/AgendaItemResearchPane/FrequentPositions/FrequentPositions.jsx [hunk 1]
```diff
@@ -18,20 +18,22 @@ const fuseOptions = {
   maxPatternLength: 32,
   minMatchCharLength: 1,
   keys: [
-    'org', 'position_number', 'position_title',
```
Comment (SidelineCory24) [inline src/Components/Agenda/AgendaItemResearchPane/FrequentPositions/FrequentPositions.jsx:21]: updating keys so search works

File: src/actions/positions.js [hunk 2]
```diff
@@ -14,27 +14,37 @@ export function positionsIsLoading(bool) {
...
         dispatch(positionsIsLoading(false));
+        dispatch(frequentPositionsIsLoading(false));
         dispatch(positionsFetchDataSuccess(response.data));
       })
       .catch(() => {
         dispatch(positionsHasErrored(true));
         dispatch(positionsIsLoading(false));
+        dispatch(frequentPositionsIsLoading(false));
       });
   };
```
Comment (scott062) [inline src/actions/positions.js:49]: Create new action that avoids loading side effects

Comment (SidelineCory24) [reply src/actions/positions.js]: done

File: src/sass/_agendaItemMaintenance.scss [hunk 3]
```diff
@@ -332,6 +332,15 @@
   }
 
   .frequent-positions-table {
+    
+    .tm-spinner-aim-fp {
+      margin-left: 33em;
+    }
```
Comment (elizabeth-jimenez) [inline src/sass/_agendaItemMaintenance.scss:338]: technically we should be able to use the same styled spinner for every tab in the research pane 🤔

Comment (elizabeth-jimenez) [reply src/sass/_agendaItemMaintenance.scss:338]: actually, check out groupLoading in AgendaItemResearchPane

File: src/actions/positions.js [hunk 4]
```diff
@@ -38,3 +52,21 @@ export function positionsFetchData(query) {
...
 }
+
+export function addFrequentPositionsData(query) {
+  return (dispatch) => {
+    dispatch(frequentPositionsIsLoading(true));
+    dispatch(frequentPositionsHasErrored(false));
+    const prefix = '/fsbid/positions';
+    api().get(`${prefix}/?${query}`)
```
Comment (elizabeth-jimenez) [inline src/actions/positions.js:61]: Why not use the existing TM frequent_positions EP?

File: src/sass/_agendaItemMaintenance.scss (hunk 3 above)
Comment (elizabeth-jimenez) [reply src/sass/_agendaItemMaintenance.scss:338]: NVM - spoke with Scott and this is for the individual loading that comes from adding the position

File: src/actions/positions.js (hunk 4 above)
Comment (SidelineCory24) [reply src/actions/positions.js:61]: I was responding to this [feedback](https://github.com/MetaPhase-Consulting/State-TalentMAP/pull/2400#discussion_r1047206472) but I can look into this ep.

**Comment by scott062** [comment]
Could we implement the loading spinner on the individual buttons instead? Thinking it'll be less distracting than the global spinner on the whole group

**Review by scott062 — APPROVED** [review]
🚀

**Pull request merged**